lib/node_modules/@stdlib/math/base/special/atanf/lib/main.js

```diff
@@ -0,0 +1,119 @@
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*
+* ## Notice
+*
+* The original C code, long comment, copyright, license, and constants are from [Cephes]{@link http://www.netlib.org/cephes}. The implementation follows the original, but has been modified for JavaScript.
+*
+* ```text
+* Copyright 1984, 1995, 2000 by Stephen L. Moshier
+*
+* Some software in this archive may be from the book _Methods and Programs for Mathematical Functions_ (Prentice-Hall or Simon & Schuster International, 1989) or from the Cephes Mathematical Library, a commercial product. In either event, it is copyrighted by the author. What you see here may be used freely but it comes with no support or guarantee.
+*
+* Stephen L. Moshier
+* moshier@na-net.ornl.gov
+* ```
+*/
+
+'use strict';
+
+// MODULES //
+
+var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
+var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
+var PI02F = require( '@stdlib/constants/float32/half-pi' );
+var PI04F = require( '@stdlib/constants/float32/fourth-pi' );
+var polyp = require( './poly_p.js' );
+
+
+// MAIN //
+
+/**
+* Computes the arctangent of a single-precision floating-point number.
+*
+* ## Method
+*
+* -   Range reduction is from four intervals into the interval from zero to  tan( pi/8 ). A polynomial approximates the function in this basic interval.
+*
+* ## Notes
+*
+* -   Relative error:
+*
+*     | arithmetic | domain  | # trials | peak    | rms     |
+*     |:-----------|:--------|:---------|:--------|:--------|
+*     | IEEE       | -10, 10 | 10^5     | 1.9e-7  | 4.1e-8  |
+*
+* @param {number} x - input value
+* @returns {number} arctangent (in radians)
+*
+* @example
+* var v = atanf( 0.0 );
+* // returns 0.0
+*
+* @example
+* var FLOAT32_PI = require( '@stdlib/constants/float32/pi' );
+*
+* var v = atanf( -FLOAT32_PI/4.0 );
+* // returns ~-0.666
+*
+* @example
+* var FLOAT32_PI = require( '@stdlib/constants/float32/pi' );
+*
+* var v = atanf( FLOAT32_PI/4.0 );
+* // returns ~0.666
+*
+* @example
+* var v = atanf( NaN );
+* // returns NaN
+*/
+function atanf( x ) {
+	var sgn;
+	var y;
+	var z;
+
+	if ( isnanf( x ) ) {
+		return NaN;
+	}
+	x = float64ToFloat32( x );
+	if ( x < 0.0 ) {
+		sgn = -1;
+		x = -x;
+	} else {
+		sgn = 1;
+	}
+	if ( x > 2.414213562373095 ) {
+		y = PI02F;
+		x = -float64ToFloat32( 1.0 / x );
+	} else if ( x > 0.4142135623730950 ) {
+		y = PI04F;
+		x = float64ToFloat32( float64ToFloat32( x - 1.0 ) / float64ToFloat32( x + 1.0 ) ); // eslint-disable-line max-len
+	} else {
+		y = 0.0;
+	}
+	z = float64ToFloat32( x * x );
+	y += float64ToFloat32( ( float64ToFloat32( polyp( z ) ) * float64ToFloat32( z * x ) ) + x ); // eslint-disable-line max-len
```

Comment:
```suggestion
	y = float64ToFloat32( y + ( float64ToFloat32( ( float64ToFloat32( polyp( z ) ) * float64ToFloat32( z * x ) ) + x ) ); // eslint-disable-line max-len
```

In-place operators need to be refactored into their assignment equivalent in order to ensure single-precision emulation; otherwise, operation result remains as double-precision.

Reply:
Tried re-adjusting tolerance after this, but anything below the current tolerance leads to failed test. So, keeping them the same.
@kgryte.

Reply:
Keeping the tolerance the same makes sense. Main thing would have been if accuracy decreased, meaning that we would have to increase the tolerance. If we didn't have to do that, should be fine.